File: mpush_reactnative_android_demo/AwesomeProject/android/app/src/main/java/com/awesomeproject/push/PopupPushActivity.java
package com.awesomeproject.push;

import android.content.Intent;
import android.os.Handler;
import android.os.Looper;
import android.os.SystemClock;
import android.util.Log;

import com.alibaba.sdk.android.push.AndroidPopupActivity;
import com.awesomeproject.MainActivity;
import com.awesomeproject.MainApplication;
import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.WritableMap;

import java.util.Map;

public class PopupPushActivity extends AndroidPopupActivity {
    @Override
    protected void onSysNoticeOpened(String title, String summary, Map<String, String> extraMap) {
        Log.d("AIXI_Push", String.format("title:%s summary:%s ext:%s", title, summary, extraMap.toString()));
        WritableMap params = Arguments.createMap();
        params.putString("title", title);
        params.putString("summary", summary);
        for (String key : extraMap.keySet()) {
            params.putString("extraMap_" + key, extraMap.get(key));
        }
        new Thread(() -> {
            SystemClock.sleep(3000);
            PushModule.sendEvent("onSysNoticeOpened", params);
        }).start();
        Intent intent = new Intent(this, MainActivity.class);
        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK);
        startActivity(intent);
        finish();
    }
}

File: mpush_android_demo/src/main/java/com/alibaba/push/android/demo/PopupPushActivity.kt
package com.alibaba.push.android.demo

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.widget.TextView
import com.alibaba.sdk.android.push.AndroidPopupActivity

class PopupPushActivity: AndroidPopupActivity() {

    private var tvMessage: TextView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_popup_push)
        tvMessage = findViewById(R.id.tv_message)
    }

    override fun onSysNoticeOpened(title: String?, content: String?, extraMap: MutableMap<String, String>?) {
        Log.d("PopupPushActivity", "title = $title, content = $content, extraMap = $extraMap")
    }

    override fun onNotPushData(intent: Intent?) {
        super.onNotPushData(intent)
        Log.d("PopupPushActivity", "onNotPushData")
    }

    override fun onParseFailed(intent: Intent?) {
        super.onParseFailed(intent)
        Log.d("PopupPushActivity", "onParseFailed")
    }
}